pkg/cmd/server/start/master_args.go

```diff
@@ -193,6 +189,14 @@ func (args MasterArgs) BuildSerializeableMasterConfig() (*configapi.MasterConfig
 		},
 	}
 
+	if args.BindAddrArg.BindAddr.URL.Scheme == "https" {
```

Comment:
Welcome to my world this past week. :)

I think I'm more inclined to use the masterURL since that will have a scheme and indicates how the user would like to interact with his master.